### ISSUE TITLE
Any plans on updating to use mysql2?

### ISSUE THREAD

**Issue by mikafri** [strongloop/loopback-connector-mysql#289]
Hi, 

We have an problem with our client who is willing to use Azure PaaS MySQL which version is 5.7, and not allowed to connect using that protocol 41, old-password thingie which is used by default mysql lib for node. 

Tested out the new mysql2 and it is supporting up-to-date authentication method, was just wondering do you have any plans to update connector to use that? 

If not, I'm probably starting to write that connector taking this project as baseline.

**Comment by ssh24** [comment]
/cc @raymondfeng 

Here is the mysql2 driver: https://github.com/sidorares/node-mysql2

**Comment by mikafri** [comment]
Yes, but do you have plans to switch loopback-connector-mysql to use mysql2? Or maybe there should be fork of this project and implement it using mysql2?

**Comment by dhmlau** [comment]
@mikafri , thanks for testing out the new mysql node module.  Is everything working the same in `mysql2` comparing to `mysql`?
Regarding Protocol 41, are you referring to this? https://simple.wikipedia.org/wiki/Protocol_41
And `mysql2` does not support Protocol 41?

**Comment by mikafri** [comment]
The problem was with Azure Paas Mysql and problem with node-mysql, it was already figured out in Azure side, and node is waiting for PR to be merged. I haven't yet started trying to take mysql2 in to use, will let you know once tested it.

**Comment by Anth0ny** [comment]
I am wanting to use the Azure PaaS MySql as well and get the same connection error. Checking to see if mysql2 has been revisited since original post. Need a solution.

**Comment by raymondfeng** [comment]
I’m open to mysql2. It seems that apis are mostly compatible as mysql. Are u guys willing to contribute a patch?

**Comment by Anth0ny** [comment]
I did some preliminary testing by simply simply replacing one reference in
mysql.js:

var mysql = require('mysql2');

That's all - no other changes, and the authentication to Azure PaaS MySql
is working. I tested basic reads and writes to the database using my
existing models.
However I am not familiar enough to be comfortable testing the connector as
thoroughly as would be needed to submit a patch myself.
